src/dataflow/src/sink/kafka.rs

```diff
@@ -143,9 +143,30 @@ struct KafkaSink {
     encoder: Encoder,
     producer: ThreadedProducer<SinkProducerContext>,
     activator: timely::scheduling::Activator,
+    txn_timeout: Duration,
 }
 
 impl KafkaSink {
+    fn transition_on_error(&self, current_state: SendState, e: KafkaError) -> SendState {
+        error!(
+            "encountered error during kafka interaction. {} in state {:?} : {}",
+            &self.name, current_state, e
+        );
+
+        match e {
+            KafkaError::Transaction(e) => {
+                if e.txn_requires_abort() {
+                    SendState::AbortTxn
+                } else if e.is_retriable() {
```

Comment:
very extremely minor: im not sure if this is spelled this way or `retryable`

Reply:
Apparently this is one of those 'made up programmer words' like 'performant' and not actually in the dictionary... It looks like retryable is the more common spelling though

Reply:
@$%!

Reply:
Welp, this is how librdkafka spells it, so I'm going to leave it as is.